services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java

```diff
@@ -292,76 +231,67 @@ public Document formulateSearchDocument(Node node, NamedAnnotations annos,
 		}
 		return document;
 	}
+	void addAnnotationsToSearchDocument(DocumentFields fields, NamedAnnotations annotations){
+		// process a map of annotation keys to values
+		Map<String, String> firstAnnotationValues = getFirsAnnotationValues(annotations);
+
+		//set the values for the document fields
+		fields.setDisease(getSearchIndexFieldValue(firstAnnotationValues, FIELD_DISEASE));
+		fields.setConsortium(getSearchIndexFieldValue(firstAnnotationValues, FIELD_CONSORTIUM));
+		fields.setTissue(getSearchIndexFieldValue(firstAnnotationValues, FIELD_TISSUE));
+		fields.setPlatform(getSearchIndexFieldValue(firstAnnotationValues, FIELD_PLATFORM));
+		try {
+			fields.setNum_samples(NumberUtils.createLong(getSearchIndexFieldValue(firstAnnotationValues, FIELD_NUM_SAMPLES)));
+		}catch (NumberFormatException e){
+			// swallow this exception, this is just a best-effort
```

Comment:
if the users did not provide a numeric value for "number of samples" then that value will not be added to the search index but is not an error.